### PR TITLE
Use PDM for modern build. Use PyTest. Add all possible type hints. 

### PR DESCRIPTION
**Pull request by jt-wang**
## What
This PR
1. uses [PDM](https://pdm.fming.dev/latest/)  to make sure users and maintainers can accurately replicate the build and run environment.
2. uses [PyTest](https://docs.pytest.org/en/7.4.x/) to wrap existing tests to have a modern testing setup, which will benefit refactoring and future development.
3. adds all possible type hints for easier maintenance and development.

## Why
`epubhv` has great potential to be the foundation on top of which other projects can develop. However, without modern build, test, and types setup, it might frustrate devs on accurately and swiftly setup and grasp the project. This PR tries to improve `epubhv` from engineering and DX (Developer Experience) aspect, and also lay a foundation for larger scale of refactoring or extension.

## Notes
There could be whitespaces added during refactoring and auto-formatting. Please feel free to dismiss them.

### CODE REVIEW

**Comment by yihong0618** [comment]
amazing and neat!
Can you help to fix the lint?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[project]
+name = "epubhv"
+version = "0.3.0"
+description = "make your epub books vertical or horizontal."
+authors = [{ name = "yihong0618", email = "zouzou0208@gmail.com" }]
+
+dependencies = [
+    "beautifulsoup4==4.12.2",
+    "bs4==0.0.1",
```
Comment (frostming) [inline pyproject.toml:9]: This package is an alias of beautifulsoup and can be removed

Comment (jt-wang) [reply pyproject.toml]: Thanks! Fixed in the latest commit. Running CI now.

**Comment by jt-wang** [comment]
> amazing and neat! Can you help to fix the lint?

@yihong0618 Sure, will do. Didn't notice ci.yaml before :(

**Comment by yihong0618** [comment]
pip install -U black
black .

**Comment by jt-wang** [comment]
> pip install -U black black .

@yihong0618 Thanks! As I turned to PDM (also thanks to @frostming for building this!), I updated `ci.yaml` to use PDM to run all scripts.

**Comment by yihong0618** [comment]
Great amazing work
Thanks a lot

**Comment by yihong0618** [comment]
Will merge tonight

**Review by yihong0618 — APPROVED** [review]

**Pull request merged**